### DISCUSSION TITLE
unexpected error

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [rustedpy/result#188] [Q&A] [answered]
Have looked at the doc but cannot see (or understand) how to capture an unexpected error and its full traceback?

Generally, on any Err() how do you capture the full traceback?

**Comment by francium** [comment, marked as answer]
There's an `as_result`, but otherwise there's no other builtin exception handling mechanism. I'd imagine most people fall back to try/catch when they're dealing with raw Python exceptions and converting them into a Result.

Here's a basic example,
```python
import random
from result import as_result, Ok

def may_fail():
    if random.random() > 0.1:
        raise RuntimeError("Something unexpected happened")
    print("Done")

@as_result(BaseException)
def do_something():
    may_fail()

r = do_something()
if isinstance(r, Ok):
    pass
else:
    print(f"Something failed: {r.err_value}")
```

**Reply by wbolster** [reply]
on top of that, exception instances have a `__traceback__` attribute
